R/check_exe.R

```diff
@@ -28,52 +53,62 @@ check_exe <- function(exe = "ss", dir = getwd(), verbose = FALSE) {
-  path_to_exe <- Sys.which(exename)[[1]]
-  # set flag for exe in path
-  exe_in_path <- path_to_exe != ""
-    
-  if (exe_in_path) {
-    # if exe is found in path
-    # make sure it has a size that makes sense for Stock Synthesis
-    # (linux systems have a command line tool called "ss" in a location
-    # like /usr/sbin/ but it's size is much smaller (about 100k vs 7MB)
-    if (file.info(path_to_exe)[["size"]] < 1e6) {
+
+  # placeholder to store path(s)
+  path_to_exe <- NULL
+  # check for exe in specified directory (or directories)
+  for (idir in seq_along(dir)) {
+    if (file.exists(file.path(dir[idir], exename))) {
+      path_to_exe[idir] <- dir[idir]
       if (verbose) {
-        message("Executable found in path that isn't Stock Synthesis:",
-          path_to_exe
-        )
+        message("Executable found in directory ", path_to_exe)
       }
-      exe_in_path <- FALSE
-    } else {
-      # normalize path and remove exe name from the end
-      # e.g. convert "C:\SS\SSB672~1.01_\ss.exe" to "C:/SS/SSv3.30.19.01_Apr15"
-      path_to_exe <- dirname(normalizePath(path_to_exe))
-      if (verbose) {
-        message("Executable found in path at ", path_to_exe)
+      # add ./ to exename so it knows to run in the current directory
+      if (.Platform[["OS.type"]] == "unix") {
+        exename <- paste0("./", exename)
       }
     }
   }
 
-  if (!exe_in_path) {
-    # if exe not found in path then check for exe in specified directory
-    # (or directories)
-    for (idir in seq_along(dir)) {
-      if (file.exists(file.path(dir[idir], exename))) {
-        path_to_exe[idir] <- dir[idir]
+  # if exe wasn't found in specified directory, look in PATH
+  if (is.null(path_to_exe)) {
+    path_to_exe <- Sys.which(exename)[[1]]
+
+    # if exe is found in PATH
+    if (path_to_exe != "") {
+      # make sure it has a size that makes sense for Stock Synthesis
+      # (linux systems have a command line tool called "ss" in a location
+      # like /usr/sbin/ but it's size is much smaller (about 100k vs 7MB)
+      if (file.info(path_to_exe)[["size"]] < 1e6) {
         if (verbose) {
-          message("Executable found in directory ", path_to_exe)
-        }
-        # add ./ to exename so it knows to run in the current directory
-        if (.Platform[["OS.type"]] == "unix") {
-          exename <- paste0("./", exename)
+          message(
+            "Executable found in PATH that isn't Stock Synthesis: ",
+            path_to_exe
+          )
         }
+        path_to_exe <- ""
       } else {
-        # if not in path or specified directory, create error
-        stop(exename, " not found in ", dir[idir], " nor in the path.")
+        if (verbose) {
+          message("Executable found in PATH at ", path_to_exe)
+        }
+        # if found in PATH and file size is large enough,
+        # then normalize path and remove exe name from the end
+        # e.g. convert "C:\SS\SSB672~1.01_\ss.exe" to "C:/SS/SSv3.30.19.01_Apr15"
```

Comment:
Are the front slashes to back slashes intentional?

Reply:
should have been double back slashes, good catch. I thought I just pasted what I got from "Sys.which('ss')", but somehow lost a slash. Fixed now.